src/main/java/seedu/omnitravel/travelactivitytypes/TravelActivityList.java

```diff
@@ -42,19 +47,41 @@ public void addTravelActivity(TravelActivity travelActivity){
     }
 
     /**
-     * Prints out all the travel activities
+     * Prints out all the travel activities.
+     * If sort is enabled, the list will be printed out from oldest to newest date.
+     * If filterDate is enabled and a date is provided, only activities from that date will be printed
+     *
+     * @param sort boolean indicating if list should be sorted
+     * @param filterDate boolean indicating if list should be filtered by date
+     * @param date the date to be filtered by
      */
-    public void listTravelActivities(){
+    public void listTravelActivities(boolean sort, boolean filterDate, LocalDate date){
+        ArrayList<TravelActivity> activities = new ArrayList<>();
+        if (filterDate) {
+            for (TravelActivity a: travelActivities) {
+                if (a.getDate().equals(date)) {
+                    activities.add(a);
+                }
+            }
+        } else {
+            activities = travelActivities;
+        }
+        if (sort) {
+            Collections.sort(activities, Comparator.comparing(TravelActivity::getDate));
+        }
         int activityCount = 0;
-        for (TravelActivity activity: travelActivities) {
+        for (TravelActivity activity: activities) {
             if (activity == null) {
                 break;
             }
             activityCount++;
             Ui.printActivity(activity, activityCount);
         }
+        if (activityCount == 0) {
+            System.out.println("There are no activities to list");
+        }
         int finalActivityCount = noOfActivities;
-        assert finalActivityCount == activityCount : "Index out of bounds while listing activities";
+        assert finalActivityCount >= activityCount : "Index out of bounds while listing activities";

```

Comment:
This assertion will cause it to fail as it will check that finalActivityCount >= activityCount. Since that is false it will cause the assertion to happen